app/src/main/java/org/projectbuendia/client/ui/SnackBar.java

```diff
@@ -0,0 +1,287 @@
+// Copyright 2015 The Project Buendia Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not
+// use this file except in compliance with the License.  You may obtain a copy
+// of the License at: http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distrib-
+// uted under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES
+// OR CONDITIONS OF ANY KIND, either express or implied.  See the License for
+// specific language governing permissions and limitations under the License.
+
+package org.projectbuendia.client.ui;
+
+import android.content.Context;
+import android.graphics.drawable.Drawable;
+import android.view.View;
+import android.view.ViewGroup;
+import android.view.animation.TranslateAnimation;
+import android.widget.ExpandableListView;
+import android.widget.ImageView;
+import android.widget.SimpleExpandableListAdapter;
+import android.widget.TextView;
+
+import org.projectbuendia.client.R;
+import org.projectbuendia.client.utils.Utils;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.TreeMap;
+
+
+public class SnackBar {
+
+    private final ViewGroup mTargetParent;
+    private final Context mContext;
+    private ExpandableListView mList;
+    private SnackBarListAdapter adapter;
+    private ArrayList groups;
+    private ArrayList children;
+    private int mMessageId;
+    private TreeMap<MessageKey, Message> mMessagesList;
+
+
+    public SnackBar(ViewGroup parent) {
+        mTargetParent = parent;
+        mContext = parent.getContext();
+
+        mMessagesList= new TreeMap<>();
+        children = new ArrayList();
+        buildList();
+    }
+
+    public void show() {
+        // TODO: discover why does not animate on the first time
+        if (mList.getVisibility() != View.VISIBLE) {
+            animate(View.VISIBLE);
+        }
+    }
+
+    public void hide() {
+        animate(View.GONE);
+    }
+
+    private void animate(int visibility) {
+        TranslateAnimation animate = new TranslateAnimation(0, 0,
+            visibility == View.VISIBLE ? mList.getHeight() : 0,
+            visibility == View.GONE ? mList.getHeight() : 0
+        );
+        animate.setDuration(700);
+        animate.setFillAfter(true);
+        mList.startAnimation(animate);
+        mList.setVisibility(visibility);
+    }
+
+    public int message(String message) {
+        return message(message, null, null, 999);
+    }
+
+    public int message(String message, int priority) {
+        return message(message, null, null, priority);
+    }
+
+    public int message(String message, String actionMessage, View.OnClickListener actionOnClick,
+                        int priority){
+        MessageKey key = new MessageKey(++mMessageId, priority);
```

Comment:
I prefer incrementing and decrementing on separate lines; it makes it easier to read and update the code. So maybe something like

    ++mMessageId;
    MessageKey key = new MessageKey(mMessageId, priority);